src/Stratis.Bitcoin/Signals/Signals.cs

```diff
@@ -42,36 +55,49 @@ public class Signals : ISignals
         /// <summary>
         /// Initializes the object with newly created instances of signalers.
         /// </summary>
-        public Signals() : this(new Signaler<Block>(), new Signaler<Transaction>())
+        public Signals() : this(new Signaler<Block>(), new Signaler<Block>(), new Signaler<Transaction>())
         {
         }
 
         /// <summary>
         /// Initializes the object with specific signalers.
         /// </summary>
-        /// <param name="blockSignaler">Signaler providing notifications about newly available blocks to its subscribers.</param>
+        /// <param name="blockConnectedSignaler">Signaler providing notifications about newly available blocks to its subscribers.</param>
+        /// <param name="blockDisonnectedSignaler">Signaler providing notifications about a block being disconnected to its subscribers.</param>
         /// <param name="transactionSignaler">Signaler providing notifications about newly available transactions to its subscribers.</param>
-        public Signals(ISignaler<Block> blockSignaler, ISignaler<Transaction> transactionSignaler)
+        public Signals(ISignaler<Block> blockConnectedSignaler, ISignaler<Block> blockDisonnectedSignaler, ISignaler<Transaction> transactionSignaler)
         {
-            Guard.NotNull(blockSignaler, nameof(blockSignaler));
+            Guard.NotNull(blockConnectedSignaler, nameof(blockConnectedSignaler));
             Guard.NotNull(transactionSignaler, nameof(transactionSignaler));
 
-            this.blocks = blockSignaler;
+            this.blocksConnected = blockConnectedSignaler;
+            this.blocksDisconnected = blockDisonnectedSignaler;
             this.transactions = transactionSignaler;
         }
 
         /// <summary>Signaler providing notifications about newly available blocks to its subscribers.</summary>
-        private ISignaler<Block> blocks { get; }
+        private ISignaler<Block> blocksConnected { get; }
+
+        /// <summary>Signaler providing notifications about blocks being disconnected to its subscribers.</summary>
+        private ISignaler<Block> blocksDisconnected { get; }
```

Comment:
@dangershony 's suggestion was that this should be ChainedHeader and not Blocks  - I have done the subscription based on Block for now but can change to ChainedHeader if you decide to take that approach.

Reply:
The intention is to replace with signalling of ChainHeaderBlock

Reply:
Yeah. Discussed that on skype- we will use ChainHeaderBlock. But after a few PRs that will change rewinding. Possibly after activation